Enable dependabot for version updates

eric-young opened the pull request:
This PR includes a dependabot configuration file that instructs dependabot to look for version updates to container images in the following charts:
- csi-vxflexos
- csm-authorization
- karavi-observability

Other charts can be updated by following a similar pattern to alter the chart values and templates, and then adding the chart to the .github/dependabot file.

Tested by running `helm template --dependency-update <chart-name>` and manually inspecting the resulting yaml for proper image specifications.

<!--
Thank you for contributing to helm-charts. Before you submit this PR we'd like to
make sure you are aware of our technical requirements and best practices:

* https://github.com/dell/helm-charts/docs/CONTRIBUTING.md
* https://helm.sh/docs/chart_best_practices/

Following our best practices right from the start will accelerate the review process and
help get your PR merged quicker.

When updates to your PR are requested, please add new commits and do not squash the
history. This will make it easier to identify new changes. The PR will be squashed
anyways when it is merged. Thanks.

For fast feedback, please @-mention maintainers that are listed in the Chart.yaml file.

Please make sure you test your changes before you push them. Once pushed, GitHub actions
will run across your changes and do some initial checks and linting. These checks run
very quickly. Please check the results. We would like these checks to pass before we
even continue reviewing your changes.
-->

#### Is this a new chart?

No

#### What this PR does / why we need it:
See above description

#### Which issue(s) is this PR associated with:

- https://github.com/dell/csm/issues/1414

#### Special notes for your reviewer:

#### Checklist:

[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]

- [ ] Chart Version bumped
- [ ] Variables are documented in the chart README.md
- [ ] Title of the PR starts with the chart name (e.g. `[charts_dir/mychartname]`) if applicable

AronAtDell commented on charts/csm-authorization/templates/proxy-server.yaml:
```diff
@@ -114,4 +114,4 @@ spec:
     protocol: TCP
     port: 8080
-    targetPort: 8080
+    targetPort: 8080
```
Newline issue.

eric-young replied:
Not really, YAML doesn't need an ending line feed, but I added it back in

AronAtDell replied:
Thanks!

AronAtDell commented:
Linting errors would be addressed for this PR?

AronAtDell commented:
How about the other charts? (Drivers/Modules). Is this an example for the owning components Teams to take forward? If yes, let's follow up with a backlog item to be completed by the respective Scrum Teams and drive to a consistent rollout for Helm.

eric-young commented:
> Linting errors would be addressed for this PR?

Linting issues are not fixed within this PR. That should be addressed separately

shaynafinocchiaro commented on .github/dependabot.yml:
```diff
@@ -0,0 +1,73 @@
+updates:
+  # github actions
+  - package-ecosystem: "github-actions"
+    target-branch: "release-v1.12.0"
```
For values that are the same for each directory (like target-branch), can we set it to a variable so it only needs to be updated in one spot?

eric-young replied:
> For values that are the same for each directory (like target-branch), can we set it to a variable so it only needs to be updated in one spot?

No, dependabot does not support that and it reads the file directly

shaynafinocchiaro replied:
Is it possible to automate the changing of the target-branch when a new release branch is created? Like another github action. Otherwise this is another thing we have to remember to maintain.

eric-young replied:
Sure, a GitHub workflow is nothing more than automation that runs in the context of a repo.

Perhaps you need an action that gets run manually that creates your new release branch, updates the config values in this dependabot file, and issues a PR as necessary.

eric-young commented:
> How about the other charts? (Drivers/Modules). Is this an example for the owning components Teams to take forward? If yes, let's follow up with a backlog item to be completed by the respective Scrum Teams and drive to a consistent rollout for Helm.

Per the commit comment, these three charts can be used as an example. Steps are laid out in the got commit text.

shaynafinocchiaro commented on charts/csi-vxflexos/values.yaml:
```diff
@@ -10,21 +10,33 @@ version: v2.11.0
   # "driver" defines the container image, used for the driver container.
-  driver: dellemc/csi-vxflexos:v2.11.0
+  driver:
+    image: dellemc/csi-vxflexos:v2.11.0
```
Would dependabot pick it up if the image were `driverImage` or `powerflexSdcImage`?

eric-young replied:
no, it does not

tdawe commented:
@eric-young Do these changes need to go into the `main` branch in order for the Dependabot checks to run? If we leave these on the feature branch, will Dependabot still run these image checks?

eric-young commented:
> @eric-young Do these changes need to go into the `main` branch in order for the Dependabot checks to run? If we leave these on the feature branch, will Dependabot still run these image checks?

I have the target-branch configured as `release-v1.12.0` so it *should* check that branch and issue PRs back to that branch

tdawe commented:
> > @eric-young Do these changes need to go into the `main` branch in order for the Dependabot checks to run? If we leave these on the feature branch, will Dependabot still run these image checks?
> 
> I have the target-branch configured as `release-v1.12.0` so it _should_ check that branch and issue PRs back to that branch

Ok, I wasn't sure if these changes needed to be under the .github/dependabot.yml on the main branch for Dependabot to actually run. If these changes remain on a feature branch (until we eventually merge to main), will Dependabot ever run? Since there are no helm chart changes, we could merge this PR directly to main if that's the case.

eric-young commented:
> > > @eric-young Do these changes need to go into the `main` branch in order for the Dependabot checks to run? If we leave these on the feature branch, will Dependabot still run these image checks?
> > 
> > 
> > I have the target-branch configured as `release-v1.12.0` so it _should_ check that branch and issue PRs back to that branch
> 
> Ok, I wasn't sure if these changes needed to be under the .github/dependabot.yml on the main branch for Dependabot to actually run. If these changes remain on a feature branch (until we eventually merge to main), will Dependabot ever run? Since there are no helm chart changes, we could merge this PR directly to main if that's the case.



> > > @eric-young Do these changes need to go into the `main` branch in order for the Dependabot checks to run? If we leave these on the feature branch, will Dependabot still run these image checks?
> > 
> > 
> > I have the target-branch configured as `release-v1.12.0` so it _should_ check that branch and issue PRs back to that branch
> 
> Ok, I wasn't sure if these changes needed to be under the .github/dependabot.yml on the main branch for Dependabot to actually run. If these changes remain on a feature branch (until we eventually merge to main), will Dependabot ever run? Since there are no helm chart changes, we could merge this PR directly to main if that's the case.

Oh, gotcha. I am not sure, honestly, if they need to be on main

My first thought would be to put it on reelase-v1.12.0, enable dependabot in the repo (I can walk someone through that), and see what happens. If dependabot does NOT see it, then we can merge JUST the dependabot.yaml file to main

eric-young commented:
> Linting errors would be addressed for this PR?

@AronAtDell  I double-checked... there are no yaml lint issues with yaml in this repo (after I fixed one that I introduced)

AronAtDell reviewed (approved):
Looks good to me! Thanks!
